### PR TITLE
[NIAK-96] Filter GET request for coursers and faculties

### PR DESCRIPTION
**Pull request by majchrow**
Just a filtering GET all request by only "the newest" data available.

### CODE REVIEW

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/CourseOfStudyService.java [hunk 1]
```diff
@@ -37,6 +37,16 @@
 
     public GetCourseOfStudiesResponse getAllResponse() {
         Set<CourseOfStudy> courseOfStudies = new HashSet<>(this.courseOfStudyRepository.findAll());
+        Set<CourseOfStudy> predecessors = new HashSet<>();
+
+        for (CourseOfStudy courseOfStudy : courseOfStudies) {
+            CourseOfStudy predecessor = courseOfStudy.getPredecessor();
+            if (predecessor != null) {
+                predecessors.add(predecessor);
+            }
```
Comment (enthyp) [inline rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/CourseOfStudyService.java:46]: Shouldn't we remove all predecessors iteratively?

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/FacultyService.java [hunk 2]
```diff
@@ -24,6 +24,17 @@
...
     public GetFacultiesResponse getAllResponse() {
         Set<Faculty> faculties = new HashSet<>(this.facultyRepository.findAll());
+        Set<Faculty> predecessors = new HashSet<>();
+
+        for (Faculty faculty : faculties) {
+            Faculty predecessor = faculty.getPredecessor();
+            if (predecessor != null) {
+                predecessors.add(predecessor);
+            }
+        }
```
Comment (enthyp) [inline rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/FacultyService.java:34]: As above

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/CourseOfStudyService.java (hunk 1 above)
Comment (enthyp) [reply rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/CourseOfStudyService.java:46]: Ah, nevermind, I get it now

Comment (majchrow) [reply rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/CourseOfStudyService.java:46]: There may be the same predecessor for multiple courses, what then? When we are adding it to the Set it actually isn't added at all, with delete statement we would need probably add another "if predecessor in courses", so I don't think it is good idea, which path do you prefer?

**Pull request merged**